implentio/ui-toolkit
# IUT-22 Create FilterChip

## Description
This component renders chips specifically for filters.

Props: "variants", "filterName"

Variants:
- "addFilter": a chip where (ex. on clicking) you can add additional filters 
- "filterValue" a chip to display currently filters being applied

filterName:
- the name of the filter. Only applicable when variant = filterValue

## Link to JIRA ticket
[Include the link here.](https://implentio.atlassian.net/jira/software/projects/IUT/boards/10?selectedIssue=IUT-22)

## Applicable Screenshots
**variant="filterValue" filterValue="test"**
default:
![image](https://github.com/implentio/ui-toolkit/assets/93560211/ed779858-4347-4dbc-8fd5-82d83522504d)
hover on button:
![image](https://github.com/implentio/ui-toolkit/assets/93560211/ea7ee5a4-1b8d-4ad3-8279-f989712ae9b1)
hover on X:
![image](https://github.com/implentio/ui-toolkit/assets/93560211/13a6e506-a650-4c16-a1b1-317990980993)

**variant="addFilter" filterValue="test"**
default:
![image](https://github.com/implentio/ui-toolkit/assets/93560211/f10b6cf0-d480-4720-a757-dd8cd1fe9011)
hover on button:
![image](https://github.com/implentio/ui-toolkit/assets/93560211/fe2ca7c2-ae62-4f1a-be63-f3b6ec35c550)

## Testing Instructions
Outline the steps required to test your changes effectively (Replace the following to what makes sense for your PR):
- Fetch and checkout my branch `git fetch && git checkout <feat/IUT-22/filterChip>`
- npm run storybook
- Check to see that the behavior follows figma
  - on hover styles change

## Checklist:

## Provide feedback:
- line 6: I have used 'text-body/small/w-medium' which is currently having a bug. To get around it I have also input its style directly as a temporary solution: 'font-medium text-leading-[1.3rem] text-tracking-[-0.01rem]' Is this ok?
- line 22: filterName is a required prop. However, its only useful and required when variant = filterValue.
